pkg/table/tables/tables.go

```diff
@@ -1188,7 +1192,10 @@ func (t *TableCommon) RemoveRecord(ctx table.MutateContext, txn kv.Transaction,
 	}
 
 	// The table has non-public column and this column is doing the operation of "modify/change column".
-	if len(t.Columns) > len(r) && t.Columns[len(r)].ChangeStateInfo != nil {
+	// DELETE will use deletable columns, which is the same as the full columns of the table.
+	// INSERT and UPDATE will only use the wrtiable columns. So they will not see column under MOIDFY/CHANGE state.
+	// This if block is for for the INSERT and UPDATE.
+	if !opt.HasIndexesLayout() && len(t.Columns) > len(r) && t.Columns[len(r)].ChangeStateInfo != nil {
```

Comment:
Please add a comment that `opt.HasIndexesLayout()` returns `true` in `DELETE` statement and `false` in `INSERT` and `UPDATE`